internal/provider/resource_vdb.go

```diff
@@ -1213,14 +1213,25 @@ func resourceVdbRead(ctx context.Context, d *schema.ResourceData, meta interface
 
 	vdbId := d.Id()
 
-	res, diags := PollForObjectExistence(func() (interface{}, *http.Response, error) {
+	res, diags, statusCode := PollForObjectExistence(func() (interface{}, *http.Response, error) {
 		return client.VDBsApi.GetVdbById(ctx, vdbId).Execute()
 	})
 
 	if diags != nil {
-		ErrorLog.Printf("Error reading the VDB, removing from state.")
-		d.SetId("")
-		return diag.Errorf("[NOT OK] Error in polling vdb")
+		_, diags, _ := PollForObjectDeletion(func() (interface{}, *http.Response, error) {
+			return client.VDBsApi.GetVdbById(ctx, vdbId).Execute()
+		})
+		// This would imply error in poll for deletion so we just log and exit.
+		if diags != nil{
+			ErrorLog.Printf("[NOT OK] Error in polling vdb")
+		}
+		// diags will be nill in case of successful poll for deletion logic aka 404
+		if diags == nil && statusCode==404{
```

Comment:
statusCode needs to be set at line 1221?

Reply:
statusCode is supposed to be set at 1221 right? instead you are using _ in third return value

Reply:
We don't need to set status code for `PollForObjectDeletion` as it polls for 404 and it being successful means that it returned 404 so we are checking for statusCode of only `PollForObjectExistence` line 1216

Reply:
Yes, but if you are not setting statusCode at 1221, how would your IF condition at line 1229 will match for "404" value?

Reply:
We are setting it on line 1216